### PR TITLE
USHIFT-132: Disable Admission Plugins for non-used APIs

### PR DESCRIPTION
**Pull request by oglok**
**Which issue(s) this PR addresses**:

Closes [USHIFT-132](https://issues.redhat.com/browse/USHIFT-132)

### CODE REVIEW

**Comment by oglok** [comment]
We need to enhance the CRD file to add the `wildcardPolicy: None` value as a default, and OpenAPI schema validation.

File: assets/crd/0000_04_routes-crd.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
```
Comment (fzdarsky) [inline assets/crd/0000_04_routes-crd.yaml:2]: Where in the OpenShift repos does this CRD come from?

**Comment by fzdarsky** [comment]
Before removing the openshift-apiserver, are we sure that we get the exact same behaviour with those CRDS + controllers as with the openshift-apiserver? I'm thinking not only about existing apps that need to run on both OpenShift and MicroShift without modification, but also, for example, if we need to add code from the policy controller for handling SCCs, will this depend on code in openshift-apiserver?

**Comment by mangelajo** [comment]
Wow, can we really remove the openshift-apiserver? what about the SCCs/etc?

File: assets/crd/0000_04_routes-crd.yaml (hunk 1 above)
Comment (oglok) [reply assets/crd/0000_04_routes-crd.yaml:2]: https://github.com/openshift/router/blob/master/deploy/route_crd.yaml

**Comment by oglok** [comment]
> 

This is something we discussed in the last team meeting. Actuaylly, the first version of the PR was basically removing the API service and adding the CRD without getting rid of the openshift apiserver. Without basic CI, it's very hard to tell.

I really want this week to get another set of eyes (@benluddy ) to help me analyse what else OpenShift API server does in the context of MicroShift. If we replace all aggregated APIs by CRDs, can we get rid of the OPenShift API server? Is there anything else regarding SCCs that the OpenShift API server is doing?

**Comment by oglok** [comment]
> Before removing the openshift-apiserver, are we sure that we get the exact same behaviour with those CRDS + controllers as with the openshift-apiserver? I'm thinking not only about existing apps that need to run on both OpenShift and MicroShift without modification, but also, for example, if we need to add code from the policy controller for handling SCCs, will this depend on code in openshift-apiserver?

That's why we need to perform a deep analysis of what the OpenShift API server does in a MicroShift context once we get rid of the aggregated APIs.

**Comment by oglok** [comment]
@Miciah has pushed this PR related to the Route CRD to OpenShift:

https://github.com/openshift/api/pull/1228

**Pull request merged**